music_assistant/server/providers/opensubsonic/sonic_provider.py

```diff
@@ -675,6 +684,38 @@ async def get_similar_tracks(self, prov_track_id: str, limit: int = 25) -> list[
         )
         return [self._parse_track(entry) for entry in songs]
 
+    async def create_playlist(self, name: str) -> Playlist:
+        """Create a new empty playlist on the server."""
+        playlist: SonicPlaylist = await self._run_async(self._conn.createPlaylist, name=name)
+        return self._parse_playlist(playlist)
+
+    async def add_playlist_tracks(self, prov_playlist_id: str, prov_track_ids: list[str]) -> None:
+        """Append the listed tracks to the selected playlist.
+
+        Note that the configured user must own the playlist to edit this way.
+        """
+        try:
+            await self._run_async(
+                self._conn.updatePlaylist, lid=prov_playlist_id, songIdsToAdd=prov_track_ids
+            )
+        except SonicError:
+            msg = f"Failed to add songs to {prov_playlist_id}, try again later."
+            raise ResourceTemporarilyUnavailable(msg)
```

Comment:
Isn't this a bit of a strange error to raise on this action ?

ResourceTemporarilyUnavailable is more for rate limiters or other reason why an action can temporary not be performed

Reply:
I put a new series up that adds a ProviderPermissionDenied error and uses that instead.